config/manager/manager.yaml

```diff
@@ -37,6 +37,17 @@ spec:
           httpGet:
             path: /healthz
             port: healthz
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - ALL
+          runAsUser: 65532
+          runAsGroup: 65532
+      securityContext:
+        runAsNonRoot: true
```

Comment:
Sadly, this will immediately kill the nice Tilt setup we have for controller testing because Tilt requires root access to the deployment.

Before we merge this, the Tiltfile somehow has to be made aware.

In fact, can you please test the tilt setup under cluster-api?

Reply:
Nice catch @Skarlso

Reply:
So I had a quick look at the CAPI Tiltfile. Indeed it would break the deployment. But I would test this nevertheless. Maybe there is magic I'm missing, but most likely not.

This is the thing that needs to be added in order to change the deployment in development:

```python
objects = decode_yaml_stream(install)
for o in objects:
    if o.get('kind') == 'Deployment' and o.get('metadata').get('name') == '<bla>':
        o['spec']['template']['spec']['securityContext']['runAsNonRoot'] = False
        break
```

Or something like that. :) Which means, in order to get this in, we require a CAPI PR that fixes the Tiltfile first.

Reply:
Thanks for  the awesome feedback!

I thought this was already covered by 
- https://github.com/kubernetes-sigs/cluster-api/pull/7846
But I will take a look into it tomorrow and also try if capa runs with tilt or requires an PR (and file that PR if so :-) )

Reply:
My god I completely forgot this existed. Why do they have ALL of that. :D :D

Reply:
To speed up start time. Interesting.

Reply:
I will check that out tomorrow and run capa with tilt to ensure I don't miss something 👍

Reply:
So to confirm: it works with the above linked change.

However: using CAPI v1.3 + tilt will not work, because the change was not cherry-picked to CAPI v1.3.

CAPI v1.4 or main works just fine.

Reply:
Uh, what a bother. But okay, this does not concern this change. I'm wondering however, how this affects upgrade procedures. I'm very well-versed in that area. What should we look out for when changing existing deployments?

Reply:
@Skarlso we could bring this up with CAPI folks.